### PR TITLE
Try reading the ca.crt and TokenReviewer JWT from the default service account

### PR DESCRIPTION
**Pull request by eh-steve**
This should help address https://github.com/hashicorp/vault-plugin-auth-kubernetes/issues/82

### CODE REVIEW

**Comment by riuvshyn** [comment]
that would be amazing!!!

**Comment by riuvshyn** [comment]
@tyrannosaurus-becks PTAL

**Comment by riuvshyn** [comment]
@kalafut any chance this change can be considered to merge-in?
This PR simplifies k8s setup alot! really looking forward to this.

**Comment by kalafut** [comment]
@riuvshyn Yes, we'll be adding it soon.

File: path_config.go [hunk 1]
```diff
@@ -97,14 +98,26 @@ func (b *kubeAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Requ
 		return logical.ErrorResponse("no host provided"), nil
 	}
 
+	localCACert, _ := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt")
+
+	localTokenReviewer, _ := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
```
Comment (jasonodonnell) [inline path_config.go:103]: These errors should not be discarded since if they don't exist, it will cause problems.

Comment (eh-steve) [reply path_config.go:103]: If the errors are nil, the byte slice will also be nil, so will be converted into an empty string later. It’s only if both the default files and the configured cert data are empty that we have an error?

Comment (jasonodonnell) [reply path_config.go:103]: That's fair!

**Comment by cyrus-mc** [comment]
@eh-steve 

I opened an issue that references this PR as the cause. 

I am unable to authenticate to external clusters (when vault is itself running in a cluster) due to the fact that in this change you default the tokenReviewerJWT to the POD token (if found) and one wasn't supplied in the auth configuration.

**Pull request merged**